homework/index.js

```diff
@@ -30,14 +30,77 @@
     return elem;
   }
 
+  function renderRepository(leftContainer, repository) {
+    leftContainer.innerHTML = '';
+    const table = createAndAppend('table', leftContainer);
+    const tBody = createAndAppend('tBody', table);
+    const trRepository = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trRepository, { text: 'Repository: ' });
+    createAndAppend('a', trRepository, {
+      text: repository.name,
+      href: repository.html_url,
+    });
+    const trDescription = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trDescription, { text: 'Description: ' });
+    createAndAppend('td', trDescription, { text: repository.description });
+    const trForks = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trForks, { text: 'Forks: ' });
+    createAndAppend('td', trForks, { text: repository.forks });
+    const trUpdated = createAndAppend('tr', tBody, { class: 'table-info' });
+    createAndAppend('td', trUpdated, { text: 'Updated: ' });
+    createAndAppend('td', trUpdated, { text: repository.updated_at.toLocaleString() });
+  }
+
+  function renderContributors(rightContainer, repository, root) {
+    rightContainer.innerHTML = '';
+    const contributorsURL = repository.contributors_url;
+    createAndAppend('p', rightContainer, { text: 'Contributions', id: 'contTitle' });
+
+    fetchJSON(contributorsURL, (err, contributors) => {
+      if (err) {
+        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
```

Comment:
In case of an error you do not want to continue executing the code that deals with rendering the contributor data. You could should add a `return` statement to return 'early' from this function in case of errors, similar to what you did in line 85.

Reply:
Hello @remarcmij code has been improved based on your comment.
Thanks.